File: app/Models/PrintJob.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class PrintJob extends Model
{
    protected $fillable = [
        'job_id', 'order_id', 'store_id', 'asset_url', 'media_size',
        'product_name', 'quantity', 'customer_name', 'settings',
        'status', 'error_message', 'printed_at',
    ];

    protected $casts = [
        'settings'   => 'array',
        'printed_at' => 'datetime',
    ];

    /**
     * Relationship to the Order.
     */
    public function order(): BelongsTo
    {
        return $this->belongsTo(Order::class);
    }

    /**
     * Relationship to the Store.
     */
    public function store(): BelongsTo
    {
        return $this->belongsTo(Store::class);
    }

    /**
     * Map this job to the structure expected by the C# Print Agent.
     */
    public function toAgentArray(): array
    {
        return [
            'JobId'        => (string) $this->job_id,
            'OrderId'      => (string) $this->order->order_number,
            'AssetUrl'     => $this->asset_url, 
            'MediaSize'    => $this->media_size,
            'Product'      => $this->product_name,
            'Quantity'     => (int) $this->quantity,
            'CustomerName' => $this->customer_name ?? 'Guest',
            'Settings'     => $this->settings ?? [
                'ColorMode'   => 'CMYK',
                'Resolution'  => 1200,
                'DuplexMode'  => 'None',
                'MediaType'   => 'Glossy',
            ],
            'ReceivedAt'   => $this->created_at->toIso8601String(),
            'State'        => 0, // 0 = Queued
        ];
    }
}

File: public/agent/QrintoPrintAgent/src/Models/PrintJob.cs
namespace QrintoPrintAgent.Models;

/// <summary>
/// Represents a print job received from the Qrinto backend.
/// </summary>
public class PrintJob
{
    public string JobId { get; set; } = string.Empty;
    public string OrderId { get; set; } = string.Empty;
    public string AssetUrl { get; set; } = string.Empty;
    public string MediaSize { get; set; } = string.Empty;
    public string Product { get; set; } = string.Empty;
    public int Quantity { get; set; } = 1;
    public string? TrayId { get; set; }
    public PrintSettings Settings { get; set; } = new();
    public string CustomerName { get; set; } = string.Empty;
    public DateTime ReceivedAt { get; set; } = DateTime.UtcNow;
    public PrintJobState State { get; set; } = PrintJobState.Queued;
    public string? ErrorMessage { get; set; }
    public string? LocalAssetPath { get; set; }
}

/// <summary>
/// Print job lifecycle states as defined in the architecture doc (Section 5.1).
/// </summary>
public enum PrintJobState
{
    Queued,
    Rendering,
    Spooled,
    Printing,
    Completed,
    Failed
}

/// <summary>
/// Per-job print settings, resolved from Admin UI defaults + tray mapping.
/// </summary>
public class PrintSettings
{
    public string ColorMode { get; set; } = "CMYK";
    public int Resolution { get; set; } = 1200;
    public string DuplexMode { get; set; } = "AutoDuplex";
    public string MediaType { get; set; } = "Glossy";
}
